tests/em/static/test_DC_Utils.py

```diff
@@ -225,27 +225,36 @@ def setUp(self):
         survey_type = ["dipole-dipole", "pole-pole", "pole-dipole", "dipole-pole"]
         data_type = "volt"
         dimension_type = "3D"
-        end_locations = np.r_[-1000.0, 1000.0, 0.0, 0.0]
+        end_locations = [
+            np.r_[-1000.0, 1000.0, 0.0, 0.0],
+            np.r_[0.0, 0.0, -1000.0, 1000.0],
+        ]
         station_separation = 200.0
         num_rx_per_src = 5
 
         # The source lists for each line can be appended to create the source
         # list for the whole survey.
         source_list = []
-        for ii in range(0, len(survey_type)):
-            source_list += utils.generate_dcip_sources_line(
-                survey_type[ii],
-                data_type,
-                dimension_type,
-                end_locations,
-                0.0,
-                num_rx_per_src,
-                station_separation,
-            )
+        lineID = []
+        for i in range(len(end_locations)):
+            for ii in range(len(survey_type)):
+                source_list += utils.generate_dcip_sources_line(
+                    survey_type[ii],
+                    data_type,
+                    dimension_type,
+                    end_locations[i],
+                    0.0,
+                    num_rx_per_src,
+                    station_separation,
+                )
```

Comment:
Another minor change here, also to make it look more Pythonic. Nothing critical, just style.
```suggestion
        for end_location in end_locations:
            for survey_type_i in survey_type:
                source_list += utils.generate_dcip_sources_line(
                    survey_type_i,
                    data_type,
                    dimension_type,
                    end_location,
                    0.0,
                    num_rx_per_src,
                    station_separation,
                )
```